specification/cdn/resource-manager/Microsoft.Cdn/stable/2017-10-12/cdn.json

```diff
@@ -2249,125 +2249,991 @@
+            "LessThanOrEqual",
+            "GreaterThan",
+            "GeaterThanOrEqual"
           ]
+        },
+        "negateCondition": {
+          "description": "Describes if this is negate condition or not",
+          "type": "boolean"
+        },
+        "matchValue": {
+          "description": "The match value for the condition of the delivery rule",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "transforms": {
+          "description": "List of transforms",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/transform"
+          }
         }
       }
-    },
-    "DeliveryRuleUrlPathCondition": {
-      "description": "Defines the URL path condition for the delivery rule.",
-      "x-ms-discriminator-value": "UrlPath",
-      "allOf": [
-        {
-          "$ref": "#/definitions/DeliveryRuleCondition"
-        }
+    },   
+    "UrlFilenameMatchConditionParameters": {
+      "description": "Defines the parameters for UrlFilename match conditions",
+      "required": [
+        "operator",
+        "matchValue",
+        "negateCondition",
+        "@odata.type"
       ],
+      "properties": {
+        "@odata.type": {
+          "type": "string",
+          "enum": [
+            "Microsoft.Azure.Cdn.Models.DeliveryRuleUrlFilenameConditionParameters"
+          ]  
+        },
+        "operator": {
+          "description": "Describes operator to be matched",
+          "type": "string",
+          "enum": [
+            "Any",
+            "Equal",
+            "Contains",
+            "BeginsWith",
+            "EndsWith",
+            "LessThan",
+            "LessThanOrEqual",
+            "GreaterThan",
+            "GeaterThanOrEqual"
+          ]
+        },
+        "negateCondition": {
+          "description": "Describes if this is negate condition or not",
+          "type": "boolean"
+        },
+        "matchValue": {
+          "description": "The match value for the condition of the delivery rule",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "transforms": {
+          "description": "List of transforms",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/transform"
+          }
+        }
+      }
+    }, 
+    "IsMobileMatchConditionParameters": {
+      "description": "Defines the parameters for IsMobile match conditions",
       "required": [
-        "parameters"
+        "operator",
+        "matchValue",
+        "negateCondition",
+        "@odata.type"
       ],
       "properties": {
-        "parameters": {
-          "description": "Defines the parameters for the condition.",
-          "$ref": "#/definitions/UrlPathConditionParameters"
+        "@odata.type": {
+          "type": "string",
+          "enum": [
+            "Microsoft.Azure.Cdn.Models.DeliveryRuleIsMobileConditionParameters"
```

Comment:
>Microsoft.Azure.Cdn.Models.DeliveryRuleIsMobileConditionParameters" [](start = 13, length = 67)

where is the definition for this?

Reply:
This is not a reference but an enum value like "GreaterThan".

Reply:
it's a reference stored in an enum, the reference for these classes @ganga is in the RP for CDN. CDN defines and published a public namespace (Microsoft.Azure.Cdn.Models) that can be reference by the public azure APIs. so you will find this class and this namespace in our CDN code, but not yet because at the moment they're just in a PR.